src/store/network.js

```diff
@@ -62,30 +64,65 @@ export default {
             let selectedNode = await AsyncCache.getSelectedNode();
-            if (!selectedNode) {
-                const network = getDefaultNetworkType();
-                selectedNode = getNodes(network)[0];
+
+            // load nodes list from statistic service
+            await dispatchAction({ type: 'network/loadNodeList' });
+
+            const networkType = getDefaultNetworkType();
+            const nodeList = networkType === 'mainnet' ? state.network.mainnetNodes : state.network.testnetNodes;
+
+            // assign node, if node list available and selectedNode is not set
+            if (nodeList.length && !selectedNode) {
+                const randomIndex = Math.floor(Math.random() * nodeList.length); //NOSONAR
+                selectedNode = nodeList[randomIndex];
+            }
+
+            // If selectedNode exists, set network
+            if (selectedNode) {
+                const network = await NetworkService.getNetworkModelFromNode(selectedNode);
+                try {
+                    const nisNodes = getNISNodes(network.type);
+                    await dispatchAction({type: 'settings/saveSetSelectedNISNode', payload: nisNodes[0]});
+                } catch {}
+                commit({ type: 'network/setGenerationHash', payload: network.generationHash });
+                commit({ type: 'network/setNetwork', payload: network.type });
+                commit({ type: 'network/setSelectedNode', payload: selectedNode });
+                commit({ type: 'network/setIsLoaded', payload: true });
+                commit({
+                    type: 'network/setSelectedNetwork',
+                    payload: network,
+                });
+                GlobalListener.setNetwork(network);
             }
-            const network = await NetworkService.getNetworkModelFromNode(selectedNode);
-            try {
-                const nisNodes = getNISNodes(network.type);
-                await dispatchAction({type: 'settings/saveSetSelectedNISNode', payload: nisNodes[0]});
-            } catch {}
-            commit({ type: 'network/setGenerationHash', payload: network.generationHash });
-            commit({ type: 'network/setNetwork', payload: network.type });
-            commit({ type: 'network/setSelectedNode', payload: selectedNode });
-            commit({ type: 'network/setIsLoaded', payload: true });
-            commit({
-                type: 'network/setSelectedNetwork',
-                payload: network,
-            });
-            GlobalListener.setNetwork(network);
             await dispatchAction({ type: 'wallet/initState' });
         },
+        loadNodeList: async ({ commit }) => {
+            try {
+                // load nodes list from statistic service
+                const [testnetNodes, mainnetNodes] = await Promise.all([
+                    NetworkService.getSelectorNodeList('testnet'),
+                    NetworkService.getSelectorNodeList('mainnet')
```

Comment:
why don't we fetch nodes based on network type?

Reply:
In mobile you have two list of nodes when you login. And you can switch from testnet to mainnet. Not sure if this answers your question :)

Reply:
The node Selector tab is showing mainnet and testnet list.
IMO we should fetch both.